File: coding-puzzles/googleChallenges/rabbit_names.js
function givePoints(letter){
  switch (letter.toLowerCase()){
    case "a":
      return 1;
    case "b":
      return 2;
    case "c":
      return 3;
    case "d":
      return 4;
    case "e":
      return 5;
    case "f":
      return 6;
    case "g":
      return 7;
    case "h":
      return 8;
    case "i":
      return 9;
    case "j":
      return 10;
    case "k":
      return 11;
    case "l":
      return 12;
    case "m":
      return 13;
    case "n":
      return 14;
    case "o":
      return 15;
    case "p":
      return 16;
    case "q":
      return 17;
    case "r":
      return 18;
    case "s":
      return 19;
    case "t":
      return 20;
    case "u":
      return 21;
    case "v":
      return 22;
    case "w":
      return 23;
    case "x":
      return 24;
    case "y":
      return 25;
    case "z":
      return 26;
  }
}

function answer(rabbitNameArr){
  var namesAndPointValues = [];
  for(i=0; i<rabbitNameArr.length; i++){
    var pointValue = 0;
    for(j=0; j<rabbitNameArr[i].length; j++){
      pointValue += givePoints( rabbitNameArr[i][j] )
    }
    namesAndPointValues[i] = [rabbitNameArr[i], pointValue];
  }

  namesAndPointValues = namesAndPointValues.sort(function(a,b){
    if(b[1] === a[1]) { console.log(b[0] + a[0]); return b[0] - a[0] }
    else { return b[1] - a[1] };
  })
  var professorFavorites = [];
  for(i=0;i<namesAndPointValues.length; i++){
    professorFavorites.push(namesAndPointValues[i][0])
  }
  return professorFavorites;
}

console.log (answer(["AL", "CJ", "Jessica"]) )

File: coding-puzzles/googleChallenges/rabbit_names.py
from operator import itemgetter, attrgetter, methodcaller

# Calculates the point value of each letter
def pointValues(let):
    return {
        'a': 1,
        'b': 2,
        "c": 3,
        "d": 4,
        "e": 5,
        "f": 6,
        "g": 7,
        "h": 8,
        "i": 9,
        "j": 10,
        "k": 11,
        "l": 12,
        "m": 13,
        "n": 14,
        "o": 15,
        "p": 16,
        "q": 17,
        "r": 18,
        "s": 19,
        "t": 20,
        "u": 21,
        "v": 22,
        "w": 23,
        "x": 24,
        "y": 25,
        "z": 26
    }[let]

# Takes in a list of names and returns the list ordered by letter point
# value and reverse alphabetically
def answer(listOfNames):
    # Declare a list for name/value pairs. For each name in the given
    # list, calculate the word point value using the pointValues
    # function for each letter. Append the name/pointValue pair to the
    # namesAndValues list
    namesAndValues = []
    for name in listOfNames:
        pointValue = 0
        for letter in name:
            pointValue += pointValues( letter.lower() )
        namesAndValues.append((name, pointValue))
    # Declare a list for the names in correct order. Sort the
    # nameAndValues list by point values (high to low) first, and then
    # reverse alphabetically for ties. Push each name to the
    # namesInOrder list and return it
    namesInOrder = []
    for nameAndValue in sorted(namesAndValues, key=itemgetter(1,0), reverse=True):
        namesInOrder.append( nameAndValue[0] )
    return namesInOrder



















